src/kit/Theme/UI.tsx

```diff
@@ -227,70 +131,171 @@ const reducer = (state: StateUI, action: ActionUI): StateUI => {
 const useUI = (): { actions: UIActions; ui: StateUI } => {
   const context = useContext(StateContext);
   if (context === undefined) {
-    throw new Error('useStore(UI) must be used within a UIProvider');
+    throw new Error('useStore(UI) must be used within a ThemeProvider');
   }
   const dispatchContext = useContext(DispatchContext);
   if (dispatchContext === undefined) {
-    throw new Error('useStoreDispatch must be used within a UIProvider');
+    throw new Error('useStoreDispatch must be used within a ThemeProvider');
   }
   const uiActions = useMemo(() => new UIActions(dispatchContext), [dispatchContext]);
   return { actions: uiActions, ui: context };
 };
 
-export const UIProvider: React.FC<{ children?: React.ReactNode; branding?: BrandingType }> = ({
-  children,
-  branding,
-}) => {
+export const ThemeProvider: React.FC<{
+  children?: React.ReactNode;
+}> = ({ children }) => {
   const [state, dispatch] = useReducer(reducer, initUI);
-  const [systemMode, setSystemMode] = useState<Mode>(() => getSystemMode());
-
-  const handleSchemeChange = useCallback((event: MediaQueryListEvent) => {
-    if (!event.matches) setSystemMode(getSystemMode());
-  }, []);
-
-  useLayoutEffect(() => {
-    // Set global CSS variables shared across themes.
-    Object.keys(globalCssVars).forEach((key) => {
-      const value = (globalCssVars as Record<RecordKey, string>)[key];
-      document.documentElement.style.setProperty(`--${camelCaseToKebab(key)}`, value);
-    });
-
-    // Set each theme property as top level CSS variable.
-    Object.keys(state.theme).forEach((key) => {
-      const value = (state.theme as Record<RecordKey, string>)[key];
-      document.documentElement.style.setProperty(`--theme-${camelCaseToKebab(key)}`, value);
-    });
-  }, [state.theme]);
-
-  // Detect browser/OS level dark/light mode changes.
+  return (
+    <StateContext.Provider value={state}>
+      <DispatchContext.Provider value={dispatch}>{children}</DispatchContext.Provider>
+    </StateContext.Provider>
+  );
+};
+
+export const UIProvider: React.FC<{
+  children?: React.ReactNode;
+  darkMode?: boolean;
+  theme: Theme;
+}> = ({ children, theme, darkMode = false }) => {
+  const [state, dispatch] = useReducer(themeStateReducer, { darkMode });
+  return (
+    <ThemeContext.Provider value={state}>
+      <ThemeDispatchContext.Provider value={dispatch}>
```

Comment:
I don't understand why we need this dispatch provider here. All it seems to do is duplicate whatever state the client is passing into `UIProvider`.

Reply:
I documented why I did this here: https://github.com/determined-ai/determined-ui/pull/10/files#diff-de4e796b62c4e1b572a4177025e78fc674477d4898bbe568e00d99f8c0d6b3ecR50

Without implementing this the `CodeEditor` for example would re-render incorrectly. When switching between light and dark mode for example it would usually re-render with white and light text on a white background still and vice-versa. 

I was not able to address this issue in any other way, and this seems like an issue that could occur with future components.

Reply:
From your comment it's not clear why you need to create additional state for the boolean instead of just passing it along via context provider along with theme.

Reply:
I believe this is addressed now that the `ThemeProvider` has been removed: https://github.com/determined-ai/determined-ui/pull/10#discussion_r1373372069

Reply:
This code is still here in my diff view?

Reply:
Okay I am missing something then. What exactly is an issue here?

Reply:
this comment made sense to me when we were passing separate contexts via the `ThemeProvider` and `UIProvider`, now it's not clear what the issue is here to me.

Reply:
What I don't understand is like #160 where you're adding state.

Reply:
Got it. The implementation now should look more correct.